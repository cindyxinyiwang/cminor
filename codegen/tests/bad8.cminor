i : integer = 1;
j : integer = i;

func: function integer (i: integer, j: integer) = {
	i = i+1;
	j = i+1;
	j = true;
	x: integer = 1;
	y: boolean;
	y = x;
	j = x;
	{
		i: integer = j;
		i > 0;
	}
}

myFunc: function void (m: boolean, n: integer) = {
	i = m +1;
	n = m++;
	m = func(1, 2);
	myFunc(2, true);
	func(i, j);
}
