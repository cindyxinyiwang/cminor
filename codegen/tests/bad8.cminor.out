i resovles to global i
i resovles to param 0
i resovles to param 0
j resovles to param 1
i resovles to param 0
j resovles to param 1
y resovles to local 1
x resovles to local 0
j resovles to param 1
x resovles to local 0
j resovles to param 1
i resovles to local 2
i resovles to global i
m resovles to param 0
n resovles to param 1
m resovles to param 0
m resovles to param 0
func resovles to global func
myFunc resovles to global myFunc
i resovles to global i
j resovles to global j
func resovles to global func
type error: cannot assign boolean to a integer
type error: cannot assign integer to a boolean
type error: cannot add boolean to a integer
type error: cannot add a boolean
type error: cannot assign integer to a boolean
type error: cannot pass variable of type integer to function myFunc
type error: cannot pass variable of type boolean to function myFunc
